### PR TITLE
[oracle] Bring back resize account

### PR DESCRIPTION
**Pull request by guibescos**
It will only pass the size check with the -Oz flag.

### CODE REVIEW

File: program/rust/src/tests/test_resize_price_account.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    // Future calls don't change the size
+    assert!(sim
+        .resize_price_account(price, &security_authority)
+        .await
+        .is_ok());
+    let price_account = sim.get_account(price).await.unwrap();
+    assert_eq!(
+        price_account.data.len(),
+        try_convert::<_, usize>(PRICE_ACCOUNT_SIZE).unwrap()
+    );
```
Comment (jayantk) [inline program/rust/src/tests/test_resize_price_account.rs:51]: try calling with a different keypair (not the security authority)

File: program/rust/src/accounts/permission.rs [hunk 2]
```diff
@@ -46,7 +46,9 @@ impl PermissionAccount {
         #[allow(clippy::match_like_matches_macro)]
         match (*key, command) {
             (pubkey, _) if pubkey == self.master_authority => true,
-            (pubkey, OracleCommand::SetMinPub) if pubkey == self.security_authority => true,
```
Comment (jayantk) [inline program/rust/src/accounts/permission.rs:49]: why did you change this minpub thing?

**Review by jayantk — COMMENTED** [review]
i'm a little confused about the SetMinPub change. aside from that, just minor comments.

File: program/rust/src/accounts/permission.rs (hunk 2 above)
Comment (guibescos) [reply program/rust/src/accounts/permission.rs:49]: Because it's not being currently used

File: program/rust/src/processor/resize_price_account.rs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        check_id(system_program.key),
+        OracleError::InvalidSystemAccount.into(),
+    )?;
+
+    // Check that it is a valid initialized price account
+    {
+        load_checked::<PriceAccount>(price_account, PC_VERSION)?;
+    }
+    let account_len = price_account.try_data_len()?;
+    if account_len < try_convert(PRICE_ACCOUNT_SIZE)? {
```
[the remark this answers is not in the document]
Comment (guibescos) [reply program/rust/src/processor/resize_price_account.rs:68]: Since it's gonna become size_of()<PriceAccountV2> let's leave it like this for this PR.
And let's merge this one after the TWAP.

File: program/rust/src/tests/pyth_simulator.rs [hunk 4]
```diff
@@ -530,6 +551,29 @@ impl PythSimulator {
...
+            bytes_of(&cmd),
+            vec![
+                AccountMeta::new(security_authority.pubkey(), true),
+                AccountMeta::new(price_account, false),
+                AccountMeta::new(system_program::id(), false),
+                AccountMeta::new_readonly(self.get_permissions_pubkey(), false),
+            ],
+        );
+
+        self.process_ixs(&[instruction], &vec![], &copy_keypair(security_authority))
```
[the remark this answers is not in the document]
Comment (guibescos) [reply program/rust/src/tests/pyth_simulator.rs:574]: Great catch, we need to use copy_keypair in some place but this is definitely not one of them.

**Review by jayantk — APPROVED** [review]
🙇
